### PR TITLE
update TiDB roadmap for CY24-CY25

### PR DESCRIPTION
**Pull request by songrijie**
### What is changed, added or deleted? (Required)

Update the TiDB roadmap for the upcoming year.  

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17225
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-roadmap.md
Comment (Oreoxmt) [inline tidb-roadmap.md:1]: Preview: https://pingcap-docsite-preview-git-preview-pi-3d0089-oreoxmts-projects.vercel.app/tidb/dev/tidb-roadmap

File: tidb-roadmap.md [hunk 1]
```diff
@@ -71,35 +110,72 @@ In the course of development, this roadmap is subject to change based on user ne
...
+            <b>Limited memory for schema meta</b><br />
+          </li>
+          <br />
+          <li>
+            <b>Distributed statistic collection</b><br />
+            Collecting statistics in parallel across multiple TiDB instances.
+          </li>
+          <br />
+          <li>
+            <b>Multi-versioned statistics多版本统计信息</b><br />
```
Comment (dveeden) [inline tidb-roadmap.md:148]:
```suggestion
            <b>Multi-versioned statistics</b><br />
```

File: tidb-roadmap.md [hunk 2]
```diff
@@ -158,82 +265,65 @@ In the course of development, this roadmap is subject to change based on user ne
...
+          <li>
+            <b>Improved dynamic privilege</b>
+            <br />Improve the dynamic privilege design and limit the implementation of Super privilege.
           </li>
           <br />
           <li>
-            <b>Column-level access control</b>
-            <br /><i>Grant and restrict access to specific columns</i>
+            <b>Marker-based log desensitization</b>
+            <br />Support marking sensitive information in the cluster log, and then you can determine whether to desensitize it according to the usage scenario.
```
Comment (qiancai) [inline tidb-roadmap.md:279]:
```suggestion
            <br />Support marking sensitive information in the cluster log. Then, you can determine whether to desensitize it according to the usage scenario.
```

File: tidb-roadmap.md [hunk 3]
```diff
@@ -158,82 +265,65 @@ In the course of development, this roadmap is subject to change based on user ne
...
           <br />
           <li>
             <b>Enhanced client-side encryption</b>
+            <br />Support client-side encryption of key fields to enhance data security.
           </li>
           <br />
           <li>
-            <b>Enhanced data masking</b>
+            <b>Dynamic desensitization of business data</b>
+            <br />Data desensitization based on different data application scenarios to ensure data security in important fields.
```
Comment (qiancai) [inline tidb-roadmap.md:319]:
```suggestion
            <br />Support desensitizing data based on different data application scenarios to ensure data security in important fields.
```

File: tidb-roadmap.md [hunk 4]
```diff
@@ -71,35 +109,73 @@ In the course of development, this roadmap is subject to change based on user ne
...
+            <b>Enhance resource group control over complex SQL</b><br />
+              Regularly assess the Request Unit (RU) consumption of complex SQL before completion of SQL execution to prevent excessively large impacts on the entire system during execution.
           </li>
           <br />
           <li>
-            <b>Runaway query control</b><br /><i>
-              An operator-controlled way to greatly enhance performance stability for workloads with unexpectedly expensive queries
-            </i>
+            <b>Automatically switch resource groups for runaway queries</b><br />
+              When a query is identified as a runaway query, users can choose to adjust it to a specific resource group and set an upper limit on resource consumption.
```
Comment (songrijie) [inline tidb-roadmap.md:131]:
```suggestion
              When a query is identified as a runaway query, users can choose to switch it to a specific resource group and set an upper limit on resource consumption.
```

File: tidb-roadmap.md [hunk 5]
```diff
@@ -110,41 +186,72 @@ In the course of development, this roadmap is subject to change based on user ne
...
           <li>
-            <b>SQL plan management</b>
-            <br /><i>Mechanism for controlling SQL plan regression</i>
+            <b>Fine-grained customization of statistics collection</b>
+            <br />Users can modify the statistics collection strategy for specific tables, such as healthiness and parallelism.
+          </li>
+          <br />
+          <li>
+            <b>Workload Repository</b>
+            <br />TiDB persists workload information in memory, including cumulative and real-time statistic data, which aids in troubleshooting and analysis.
```
Comment (songrijie) [inline tidb-roadmap.md:218]:
```suggestion
            <br />TiDB persists workload information in memory to permanent volume, including cumulative and real-time statistic data, which aids in troubleshooting and analysis.
```

File: tidb-roadmap.md [hunk 6]
```diff
@@ -110,41 +186,72 @@ In the course of development, this roadmap is subject to change based on user ne
...
           </li>
         </ul>
       </td>
       <td>
         <ul>
           <li>
-            <b>Materialized views</b>
-            <br /><i>Store pre-computed results as a persistent data view to boost query performance</i>
+            <b>Workload analysis</b>
+            <br />Analyze historical workload data from the Workload Repository and provide optimization recommendations, such as SQL tuning and adjustments to statistics collection strategies.
```
Comment (songrijie) [inline tidb-roadmap.md:246]:
```suggestion
            <br />Analyze historical workload data from the Workload Repository and provide optimization recommendations, such as SQL tuning and statistics collection.
```

**Review by Oreoxmt — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Pull request merged**